psh/echo/echo.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Phoenix-RTOS
+ *
+ * echo - display a line of text
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include "../psh.h"
+
+
+static void psh_echoinfo(void)
+{
+	printf("display a line of text");
+}
+
+
+static void psh_echo_help(const char *prog)
+{
+	printf("Usage: %s [options] [string]\n", prog);
+	printf("  -h:  shows this help message\n");
+}
+
+
+static size_t psh_echo_printVar(const char *var)
+{
+	size_t i = 1;
```

Comment:
This produces an error. In the loop in line 77 we already check if we are on the end of string. Passing command line:
```
echo $ abcd
```
causes program to skip end of string as `1` is returned and print whatever is after that. As in presented example we print `abcd` and then, in mentioned loop skip to it again (as valid call to next `argv`) and print it again causing wrong behavior (double print of the next argument). Also executing command:
```
echo $ $
```
sometimes produces an output of
```
P P
```
which means that some uninitialized, or random memory is printed.

![Screenshot from 2022-02-23 20-27-50](https://user-images.githubusercontent.com/80880689/155393454-5d7e70c5-f412-4974-80a6-59a9ea7350c4.png)

Reply:
Good find, glad I added you to reviewers ;)